packages/itwin/ec3-widget/src/components/ExportModal.tsx

```diff
@@ -0,0 +1,206 @@
+
+interface ExportProps {
+  projectName: string;
+  isOpen: boolean;
+  close: () => void;
+  templateId: string | undefined;
+  token: string | undefined;
+}
+
+const ExportModal = (props: ExportProps) => {
+  const PIN_INTERVAL = 1000;
+  const ec3JobsClient = useEC3JobsClient();
+
+  const [jobStatus, setJobStatus] = useState<CarbonUploadState>();
+  const [jobLink, setJobLink] = useState<Link>();
+
+  const intervalRef = useRef<number>();
+
+  const pinStatus = useCallback(
+    (job: EC3Job) => {
+      const intervalId = window.setInterval(async () => {
+        const token =
+          (await IModelApp.authorizationClient?.getAccessToken()) ?? "";
+        if (job.id && token) {
+          const currentJobStatus =
+            await ec3JobsClient.getEC3JobStatus(token, job.id);
+          if (currentJobStatus.status) {
+            if (
+              currentJobStatus.status === CarbonUploadState.Succeeded
+            ) {
+              setJobLink(currentJobStatus._links.ec3Project);
+            }
+            setJobStatus(currentJobStatus.status);
+          } else {
+            setJobStatus(CarbonUploadState.Failed);
+            toaster.negative("Failed to get job status. 😔");
+          }
+        }
+      }, PIN_INTERVAL);
+      intervalRef.current = intervalId;
+    },
+    [setJobLink, setJobStatus, ec3JobsClient]
+  );
+
+  const runJob = useCallback(
+    async (token: string) => {
+      const accessToken =
+        (await IModelApp.authorizationClient?.getAccessToken()) ?? "";
+      if (props.templateId && token) {
+        try {
+          const jobRequest: EC3JobCreate = {
+            configurationId: props.templateId,
+            projectName: props.projectName,
+            ec3BearerToken: token,
+          }
+          const jobCreated = await ec3JobsClient.createJob(
+            accessToken,
+            jobRequest
+          );
+          if (jobCreated.id) {
+            pinStatus(jobCreated);
+          } else {
+            setJobStatus(CarbonUploadState.Failed);
+            toaster.negative("Failed to create EC3 job. 😔");
+          }
+        } catch (e) {
+          setJobStatus(CarbonUploadState.Failed);
+          toaster.negative("You do not have the required permissions. Please contact the project administrator.");
+          /* eslint-disable no-console */
+          console.error(e);
+        }
+      } else {
+        setJobStatus(CarbonUploadState.Failed);
+        toaster.negative("Invalid reportId.");
+      }
+    },
+    [props, pinStatus, ec3JobsClient]
+  );
+
+  const onClose = useCallback(() => {
+    setJobStatus(undefined);
+    setJobLink(undefined);
+    window.clearInterval(intervalRef.current);
+    props.close();
+  }, [props]);
+
+  const getStatusComponent = useCallback(
+    (status: CarbonUploadState, link: string | undefined) => {
+      switch (status) {
+        case CarbonUploadState.Queued:
+          return (
+            <div className="ec3w-progress-radial-container">
```

Comment:
Move all these "status components" into their own separate files in a subdirectory.

Reply:
will wait to do this until a decision is made about the export modal

Reply:
Can be done after initial release.